File: solutions/12.java
public class 12 {
    public static void main(String[] a){
        int num=44;
        String ans=intToRoman(num);
        System.out.println(ans);
        
    }
    static  String intToRoman(int num) {
        int[] nums={1,4,5,9,10,40,50,90,100,400,500,900,1000};
        String[] sym={"I","IV","V","IX","X","XL","L","XC","C","CD","D","CM","M"};
        int i=12;
        String op="";
        for(i=12;i>=0;i--){
            while(num>=nums[i]){
               op +=sym[i];
                num -=nums[i];
            }
        }
        return op;
    }
    
}

File: solutions/12.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  string intToRoman(int num) {
    // 1000, 2000, 3000
    string M[] = {"", "M", "MM", "MMM"};
    // 100, 200, 300, .. 900
    string C[] = {"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
    // 10, 20, ... 90
    string X[] = {"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
    // 1, 2, ... 9
    string I[] = {"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
    return M[num / 1000] + C[(num % 1000) / 100] + X[(num % 100) / 10] +
           I[num % 10];
  }
};
